tx_clients/utils/web.py

```diff
@@ -0,0 +1,139 @@
+import base64
+import cStringIO
+import json
+import warnings
+
+from zope.interface import implements
+
+from twisted.web import client
+from twisted.web.iweb import IBodyProducer
+from twisted.internet import defer, protocol
+from twisted.internet.task import cooperate
+from twisted.web.http import PotentialDataLoss
+
+
+class StringBodyProducer(client.FileBodyProducer):
```

Comment:
I don't have a strong opinion, but recently looking through some Ygg stuff, I found a file that has a ton of different stuff in it, which lead to some circular dependencies.  What do you think about breaking things of a particular type/class out into separate files?  This isn't big now, but don't want it to become a dumping ground (although, the temptation/need will certainly be less than in Ygg).